data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/eventtype/EventTypeCreatorImpl.java

```
@@ -0,0 +1,114 @@
+
+    private static final Integer DNS1123_SUBDOMAIN_MAX_LENGTH = 253;
+
+    private final MixedOperation<EventType, KubernetesResourceList<EventType>, Resource<EventType>> eventTypeClient;
+
+    private final Lister<EventType> eventTypeLister;
+
+    private MessageDigest messageDigest;
+
+    private final WorkerExecutor executor;
+
+    private static final Logger logger = LoggerFactory.getLogger(EventTypeCreatorImpl.class);
+
+    public EventTypeCreatorImpl(
+            MixedOperation<EventType, KubernetesResourceList<EventType>, Resource<EventType>> eventTypeClient,
+            Lister<EventType> eventTypeLister,
+            Vertx vertx) {
+        this.eventTypeClient = eventTypeClient;
+        this.eventTypeLister = eventTypeLister;
+        this.executor = vertx.createSharedWorkerExecutor("et-creator-worker", 1);
+        try {
+            this.messageDigest = MessageDigest.getInstance("MD5");
+        } catch (NoSuchAlgorithmException ignored) {
+            this.messageDigest = null;
+        }
+    }
+
+    private String getName(CloudEvent event, DataPlaneContract.Reference reference) {
+        final var suffixString = event.getType() + event.getSource() + reference.getNamespace() + reference.getName();
+        this.messageDigest.reset();
+        this.messageDigest.update(suffixString.getBytes());
+        final var suffix = Hex.encodeHexString(this.messageDigest.digest());
+        final var name = String.format("et-%s-%s", reference.getName(), suffix).toLowerCase();
+        if (name.length() > DNS1123_SUBDOMAIN_MAX_LENGTH) {
+            return name.substring(0, DNS1123_SUBDOMAIN_MAX_LENGTH);
+        }
+        return name;
+    }
+
+    private EventType eventTypeExists(String etName, DataPlaneContract.Reference reference) {
+        if (this.eventTypeLister == null) {
+            logger.debug("no eventtype lister, creating eventtype and might run into failure");
+            return null;
+        }
```

Should we fail fast here in the constructor?